asmcomp/asmgen.ml

```diff
@@ -167,109 +131,50 @@ let compile_unit _output_prefix asm_filename keep_asm
        if create_asm && not keep_asm then remove_file asm_filename
     )
 
-let set_export_info (ulambda, prealloc, structured_constants, export) =
-  Compilenv.set_export_info export;
-  (ulambda, prealloc, structured_constants)
-
 let end_gen_implementation ?toplevel ~ppf_dump
-    (clambda:clambda_and_constants) =
+    (clambda : Clambda.with_constants) =
   Emit.begin_assembly ();
+  Cmmgen.reset ();
```

Comment:
Why was that required ? Ins't the one at the beginning of compile_fundecl sufficient ?

By the way, by quickly looking at it, no reset would be required if `Cmmgen_state.constants` and `data_items` where cleaning their state.

Reply:
I've fixed things so `reset` is not needed.